lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -87,6 +87,8 @@ def _check_rh_versions(self, pkg_mgr_name, collected_facts):
             # can define a provider to either the package or yum action plugins.
             if int(collected_facts['ansible_distribution_major_version']) < 8:
                 pkg_mgr_name = 'yum'
+            elif (collected_facts['ansible_distribution']) == 'Amazon':
```

Comment:
This does not look like a right place to add the special case, I would move the conditional one level up, so something like:
```python
            if collected_facts['ansible_distribution'] == 'Fedora':
               ...
            elif collected_facts['ansible_distribution'] == 'Amazon':
               pkg_mgr_name = 'yum'
            else:
               # If it's not Fedora or Amazon and it's Red Hat family of distros, assume RHEL
               ...
```

Reply:
thanks for the feedback, I've rewritten this block to use the recommended pattern

Reply:
Sorry I wasn't being clear. What I meant was the following:
```diff
diff --git a/lib/ansible/module_utils/facts/system/pkg_mgr.py b/lib/ansible/module_utils/facts/system/pkg_mgr.py
index a7dec44457..24c9dbb88b 100644
--- a/lib/ansible/module_utils/facts/system/pkg_mgr.py
+++ b/lib/ansible/module_utils/facts/system/pkg_mgr.py
@@ -78,8 +78,10 @@ class PkgMgrFactCollector(BaseFactCollector):
                 # If there's some new magical Fedora version in the future,
                 # just default to dnf
                 pkg_mgr_name = 'dnf'
+        elif collected_facts['ansible_distribution'] == 'Amazon':
+            pkg_mgr_name = 'yum'
         else:
-            # If it's not Fedora and it's Red Hat family of distros, assume RHEL
+            # If it's not one of the above and it's Red Hat family of distros, assume RHEL
             # or a clone. For versions of RHEL < 8 that Ansible supports, the
             # vendor supported official package manager is 'yum' and in RHEL 8+
             # (as far as we know at the time of this writing) it is 'dnf'.
```

Reply:
@mkrizek completely understood now, I had to step back and look at the whole _check_rh_versions instead of the else block was was looking at. Thanks!